src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -88,7 +88,7 @@ TransactionManager::mount_ertr::future<> TransactionManager::mount()
     scanner.get_segment_managers());
   return segment_cleaner->init_segments().safe_then(
     [this](auto&& segments) {
-    return journal->replay(
+    return journal->as_seg_journal().replay(
```

Comment:
All journals need to be able to replay, why cast to SegmentedJournal first?

Reply:
I added a commit to generalize replay().